java/compression/src/test/java/org/apache/arrow/compression/TestArrowReaderWriterWithCompression.java

```diff
@@ -79,11 +119,130 @@ public void testArrowFileZstdRoundTrip() throws Exception {
            new ArrowFileReader(new ByteArrayReadableSeekableByteChannel(out.toByteArray()),
              allocator, NoCompressionCodec.Factory.INSTANCE)) {
       Assert.assertEquals(1, reader.getRecordBlocks().size());
+      Exception exception = Assert.assertThrows(IllegalArgumentException.class, () -> reader.loadNextBatch());
+      String expectedMessage = "Please add arrow-compression module to use CommonsCompressionFactory for ZSTD";
+      Assert.assertEquals(expectedMessage, exception.getMessage());
+    }
+  }
+
+  @Test
+  public void testArrowFileZstdRoundTripWithDictionary() throws Exception {
```

Comment:
Do you think we still need the original test function `testArrowFileZstdRoundTrip`? Is this new test case possibly testing the same code path + dictionaries?

Reply:
It is the same path + dictionaries, I refactored and reorganized the test cases, does it make sense or useful? I think my previous code had duplication.